tests/codegen/float/f128.rs

```diff
@@ -60,7 +63,8 @@ pub fn f128_le(a: f128, b: f128) -> bool {
     a <= b
 }
 
-// x86-LABEL: void @f128_neg({{.*}}sret([16 x i8])
+// x86-nosse-LABEL: void @f128_neg({{.*}}sret([16 x i8])
+// x86-sse-LABEL: <16 x i8> @f128_neg(fp128
```

Comment:
huh...

this... feels incorrect? yet I see how it isn't. interesting.

Reply:
@tgross35 this is the correct representation, then? as a vector, to guarantee passing in xmm registers?

Reply:
From the ABI

> Arguments of types __float128, _Decimal128 and __m128 are split into two halves. The least significant ones belong to class SSE, the most significant one to class SSEUP

So passing them in the same way as __m128 seems reasonable.

Not needed here, but these tests should probably be split. I originally added them to verify we do correct lowering for operations (before they were supported enough to run tests without crashing, but still reasonably useful), but didn't really intend to check the calling convention with it. So this file could become target-agnostic and drop all types from `LABEL`, then add a separate test to check passing and returning.

Reply:
Our `PassMode` just lets us say "vector type of this size", we can't distinguish `<16 x i8>` from `<2 x i64>` from `<2 x f64>`. I guess we implicitly assume LLVM passes all of them the same way. So if that's not the case we have a more fundamental problem.